streams.go

```diff
@@ -127,28 +127,25 @@ func (w *baseStreamWriter[T]) Close() {
 	}
 }
 
-func (w *baseStreamWriter[T]) writeStreamBytes(data []byte, onDone *func()) {
+func (w *baseStreamWriter[T]) writeStreamBytes(chunks [][]byte, onDone *func()) {
 	w.lock.Lock()
 	chunkIndex := w.chunkIndex
 
-	for i := 0; i < len(data) && !w.closed.Load(); i += STREAM_CHUNK_SIZE {
-		end := i + STREAM_CHUNK_SIZE
-		if end > len(data) {
-			end = len(data)
-		}
+	for i := 0; i < len(chunks) && !w.closed.Load(); i++ {
+		chunk := chunks[i]
 
 		w.engine.waitForBufferStatusLow(protocol.DataPacket_RELIABLE)
 
 		if err := w.engine.publishStreamChunk(&protocol.DataStream_Chunk{
 			StreamId:   w.streamId,
-			Content:    data[i:end],
+			Content:    chunk,
 			ChunkIndex: chunkIndex,
 		}, w.destinationIdentities); err != nil {
 			w.engine.log.Errorw("could not publish stream chunk", err)
 		}
 
 		if w.onProgress != nil && w.totalSize != nil {
-			progress := float64(i) / float64(*w.totalSize)
+			progress := float64(len(chunk)) / float64(*w.totalSize)
 			(*w.onProgress)(progress)
```

Comment:
Don't think you need to de-ref here, you can just call `w.onProgress(progress)` I believe.

Reply:
It wasn't allowing me to do that